crates/core/src/transport/connection_handler.rs

```diff
@@ -0,0 +1,358 @@
+use super::*;
+use crate::node::PeerId;
+use aes_gcm::{aes::Aes128, KeyInit};
+use futures::channel::oneshot;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::vec::Vec;
+use std::{borrow::Cow, time::Duration};
+use tokio::net::UdpSocket;
+use tokio::sync::mpsc;
+use tokio::task;
+
+use super::{
+    connection_info::ConnectionInfo,
+    crypto::{TransportKeypair, TransportPublicKey},
+};
+
+/// The maximum size of a received UDP packet, MTU typically is 1500
+pub(super) const MAX_PACKET_SIZE: usize = 1500;
+
+pub(super) type ConnectionHandlerMessage = (SocketAddr, Vec<u8>);
+
+const PROTOC_VERSION: [u8; 2] = 1u16.to_le_bytes();
+
+pub struct PeerConnection {
+    recv_inbound: mpsc::Receiver<PacketData>,
+    send_outbound: mpsc::Sender<PacketData>,
+}
+
+impl PeerConnection {
+    pub async fn recv(&self) -> Result<Vec<u8>, ConnectionError> {
+        todo!()
+    }
+
+    pub async fn send(&self, _message: Vec<u8>) -> Result<(), ConnectionError> {
+        todo!()
+    }
+}
+
+pub(crate) struct ConnectionHandler {
+    connection_info: HashMap<PeerId, ConnectionInfo>,
```

Comment:
I see `PeerId` references a libp2p peer id, how much are we relying on libp2p at this point? It seems like it might only add complexity relative to just using a `SocketAddr`, at least within the `transport` module.

Reply:
At the end of this we probably are getting rid of the PeerId and the libp2p deps. Just ignore, since still is wip